cmd/snapctl/task.go

```diff
@@ -114,14 +114,226 @@ func stringValToInt(val string) (int, error) {
+		// new schedule window)
+		stop = t.Schedule.StopTime
+	} else if start == nil && stop != nil {
+		// otherwise, if only the stop date/time was specified, we will use it to replace the current schedule's
+		// stop date/time; so for now we simply grab the current schedule's start time (for use later in checking
+		// the boundaries of the new schedule window)
+		start = t.Schedule.StartTime
+	}
+	// check to make sure that our new time window is a valid schedule
+	if err := t.checkScheduleWindow(*start, *stop); err != nil {
+		return err
+	}
+	// if it's a valid window, then use the new start/stop values to set the
+	// start/stop date/times for our new schedule
+	t.Schedule.StartTime = start
+	t.Schedule.StopTime = stop
+	// and return a nil error (indicating success)
+	return nil
+}
+
+// parse the command-line options and use them to setup a new schedule for this task
+func (t *task) setScheduleFromCliOptions(ctx *cli.Context) error {
+	// if there is no schedule associated with this task, create an empty one
+	if t.Schedule == nil {
+		t.Schedule = &client.Schedule{}
+	}
+	// check the start, stop, and duration values to see if we're looking at a windowed schedule (or not)
+	// first, get the parameters that define the windowed schedule
+	start := mergeDateTime(
+		strings.ToUpper(ctx.String("start-time")),
+		strings.ToUpper(ctx.String("start-date")),
+	)
+	stop := mergeDateTime(
+		strings.ToUpper(ctx.String("stop-time")),
+		strings.ToUpper(ctx.String("stop-date")),
+	)
+	// Grab the duration string (if one was passed in) and parse it
+	durationStr := ctx.String("duration")
+	var duration *time.Duration
+	if ctx.IsSet("duration") || durationStr != "" {
+		d, err := time.ParseDuration(durationStr)
+		if err != nil {
+			return fmt.Errorf("Bad duration format; %v", err)
+		}
+		duration = &d
+	}
+	// Grab the interval for the schedule (if one was provided); we will use this
+	// string to determine the type of schedule ('simple' or 'cron') and to update
+	// the interval of the existing schedule (note that an interval value was
+	// not passed in and there is no interval defined for the schedule associated
+	// with this task, it's an error)
+	interval := ctx.String("interval")
+	if !ctx.IsSet("interval") && interval == "" && t.Schedule.Interval == "" {
+		return fmt.Errorf("Usage error (missing interval value); when constructing a new task schedule, an interval must be provided")
+	}
+	// if a start, stop, or duration value was provided, then the CLI options are being used to
+	// specify a new 'windowed' schedule for this task, so use them to replace the 'windowed' schedule
+	// in the current task (if one exists); note that it is an error to try to replace an existing
+	// sechedule with a new schedule of a different type, so throw an error if that is the case
+	if start != nil || stop != nil || duration != nil {
+		t.setWindowedSchedule(start, stop, duration)
```

Comment:
`setWindowedSchedule` is returning an error, but that is not being captured here. This should return that error `return t.setWindowedSchedule(start, stop, duration), and remove the `else if`.